components/brave_service_keys/service_key_utils.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_service_keys/service_key_utils.h"
+
+#include <vector>
+
+#include "base/base64.h"
+#include "brave/components/brave_service_keys/buildflags.h"
+#include "crypto/hmac.h"
+#include "crypto/sha2.h"
+
+namespace brave_service_keys {
+
+namespace {
+
+constexpr char kDigest[] = "digest";
+constexpr char kAuthorization[] = "authorization";
+
+}  // namespace
+
+std::pair<std::string, std::string> GetDigestHeader(
+    const std::string& payload) {
+  const std::string value = base::StrCat(
+      {"SHA-256=", base::Base64Encode(crypto::SHA256HashString(payload))});
+  return std::make_pair(kDigest, value);
+}
+
+std::optional<std::pair<std::string, std::string>> GetAuthorizationHeader(
+    const std::string& service_key,
+    base::span<const std::pair<std::string, std::string>> headers) {
```

Comment:
I think we should add the `full_url` for the request and the list of headers to be signed separately from the actual header key/value pairs.